tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc

```diff
@@ -4448,10 +4450,28 @@ class GenericConvertUnsortedSegmentReductionOp : public OpRewritePattern<OpTy> {
     // Broadccast the initial value for reduction. This will become the
     // 'operand' parameter to scatter to for the final scatter op.
     Value init = ConcreteClass::GetInitialValue(data_type.getElementType(),
-                                                op.getLoc(), &rewriter);
-    auto broadcasted_init = rewriter.create<mhlo::BroadcastOp>(
-        op.getLoc(), output_type, init,
-        GetI64ElementsAttr(output_shape, &rewriter));
+                                                loc, &rewriter);
+    auto data_shape = data_type.getShape();
```

Comment:
Please avoid `auto` when the type isn't obvious from the context (`ArrayRef<int64_t>`) here.

Reply:
> Please avoid `auto` when the type isn't obvious from the context (`ArrayRef<int64_t>`) here.

fixed